kratos/input_output/unv_output.h

```diff
@@ -0,0 +1,235 @@
+    // # R.  4: user-specified text (id_lines_1_to_5[0])
+    // # R.  5: user-specified text (id_lines_1_to_5[1])
+    // # R.  6: user-specified text (id_lines_1_to_5[2])
+    // # R.  7: user-specified text (id_lines_1_to_5[3])
+    // # R.  8: user-specified text (id_lines_1_to_5[4])
+    // # R.  9: (model_type) (analysis_type) (data_characteristic) (result_type) (data_type) (nvaldc)
+    // # R. 10: (design_set_id) (iteration_number) (solution_set_id) (boundary_condition) (load_set) (mode_number) (time_stamp_number) (frequency_number)
+    // # R. 11: (creation_option) (Unknown)*7
+    // # R. 12: (time) (frequency) (eigenvalue) (nodal_mass) (viscous_damping_ratio) (hysteretic_damping_ratio)
+    // # R. 13: (eigenvalue_re) (eigenvalue_im) (modalA_re) (modalA_im) (modalB_re) (modalB_im)
+
+    void WriteNodalResultValues(std::ofstream &outputFile, const Node<3>& node, const Variable<bool>& rVariable);
+    void WriteNodalResultValues(std::ofstream &outputFile, const Node<3>& node, const Variable<int>& rVariable);
+    void WriteNodalResultValues(std::ofstream &outputFile, const Node<3>& node, const Variable<double>& rVariable);
+    void WriteNodalResultValues(std::ofstream &outputFile, const Node<3>& node, const Variable<array_1d<double,3>>& rVariable);
+    void WriteNodalResultValues(std::ofstream &outputFile, const Node<3>& node, const Variable<Vector>& rVariable);
+    void WriteNodalResultValues(std::ofstream &outputFile, const Node<3>& node, const Variable<Matrix>& rVariable);
+
+    template<class TVariablebleType>
+    int GetUnvVariableName(const TVariablebleType& rVariable) {
+        if(rVariable == VELOCITY) return 11;
+        if(rVariable == TEMPERATURE) return 5;
+        if(rVariable == PRESSURE) return 117;
+
+        return 1000;
+    }
+
+    template<class TVariablebleType>
+    void WriteNodalResultRecords(const TVariablebleType& rVariable, const int numComponents, const double timeStep) {
+        std::ofstream outputFile;
+        outputFile.open(mOutputFileName, std::ios::out | std::ios::app);
+
+        const int dataSetNumberForResults = 2414;
+        std::string dataSetName = "NodalResults";
+        std::string dataSetLabel = rVariable.Name();
+
+        outputFile << std::setw(6)  << "-1" << "\n";                                                // Begin block
+        outputFile << std::setw(6)  << dataSetNumberForResults << "\n";                             // DatasetID
+
+        outputFile << std::setw(10) << dataSetLabel << "\n";                                        // Record 1 - Label
+        outputFile << std::setw(6)  << dataSetName << "\n";                                         // Record 2 - Name
+        outputFile << std::setw(10) << as_integer(DatasetLocation::DATA_AT_NODES) << "\n";          // Record 3
+
+        // String records, seems like you can put anything you want.
+        outputFile << "Test1" << "\n";                                                               // Record 4
```

Comment:
This should be removed?

Reply:
No, it has to be something, I can change it so its only a "\n"